CMakeLists.txt

```diff
@@ -1,4 +1,8 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
```

Comment:
I did it to avoid this warning:
```
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   websockets_shared
```
Also it's better to depend on latest libraries/tools, so when you need something from newer version nobody will say that this newer version is not available on some platforms :) Do you have an example of platform that doesn't have CMake 3.0?